### DISCUSSION TITLE
Retrieving the complete assets for an Entry

### DISCUSSION THREAD

**Discussion by 321zeno** [statamic/cms#10954] [Q&A]
Apologies if this has been answered before, I couldn't find an answer here or in the docs.

I'm using the Eloquent driver, and Blade views.

I've got an entry that contains some assets for an image gallery.

I'm retrieving it by slug in a Laravel controller
```php
$entry = Entry::query()
  ->where('collection', 'items')
  ->where('slug', $slug)
  ->firstOrFail()
```
When I `dd($entry->image_gallery)` I get
```
array:4 [▼ // app/Http/Controllers/ItemsController.php:42
  0 => "image2.jpg"
  1 => "image2.jpg"
  2 => "image1.jpg"
  3 => "image1.jpg"
]
```
How can I retrieve the "full" asset fields, with objects that have the permalink, alt attributes, etc?

**Comment by 321zeno** [comment]
Just realised that the EntryModel returned by `firstOrFail()` does not have the `Augmentable` methods.
If I change it to `first()`
```php
$entry = Entry::query()
  ->where('collection', 'items')
  ->where('slug', $slug)
  ->first()
```
Then I am able to call
`dd($item->toEvaluatedAugmentedArray('image_gallery'))` which gives me the whole asset as an array but it would be good to know if this is the best practice

**Comment by jasonvarga** [comment]
`$entry->image_gallery` should be giving you a collection of asset instances.

If you're only seeing an array of filenames, maybe you got rid of or renamed the blueprint field? The data would remain, but since there's no corresponding blueprint field it wouldn't know to convert the filenames to assets.
